packages/ingest/test/test-granule.js

```diff
@@ -666,6 +666,158 @@ test('ingestFile throws error when configured to handle duplicates with error',
   t.is(error.message, `${destFileKey} already exists in ${destBucket} bucket`);
 });
 
+test('ingestFile skips ingest when duplicateHandling is skip', async (t) => {
+  const sourceBucket = t.context.internalBucket;
+  const destBucket = t.context.destBucket;
+  const file = {
+    path: randomString(),
+    name: 'test.txt'
+  };
+  const key = path.join(file.path, file.name);
+  const params = { Bucket: sourceBucket, Key: key, Body: randomString(30) };
+  await s3PutObject(params);
+
+  const duplicateHandling = 'skip';
+  const fileStagingDir = 'file-staging';
+  const testGranule = new TestS3Granule(
+    {},
+    collectionConfig,
+    {
+      host: sourceBucket
+    },
+    fileStagingDir,
+    false,
+    duplicateHandling,
+  );
+
+  const oldfiles = await testGranule.ingestFile(file, destBucket, duplicateHandling);
+  t.is(oldfiles.length, 1);
+  t.is(oldfiles[0].duplicate_found, undefined);
+  t.is(oldfiles[0].fileSize, params.Body.length);
+
+  // update the source file with different content and ingest again
+  params.Body = randomString(100);
+  await s3PutObject(params);
+  const newfiles = await testGranule.ingestFile(file, destBucket, duplicateHandling);
+  t.is(newfiles.length, 1);
+  t.true(newfiles[0].duplicate_found);
+  t.is(newfiles[0].fileSize, oldfiles[0].fileSize);
+  t.not(newfiles[0].fileSize, params.Body.length);
+});
+
+test('ingestFile replaces file when duplicateHandling is replace', async (t) => {
```

Comment:
Correct me if I'm wrong, but this is testing the case where no `cksum` or `filesize` are provided. Should there be another test that validates the functionality of a file with the same `cksum` with `replace` duplicate handling being left alone?

Reply:
Checksumming only occurs in the `version` case, not the `replace` case, as there is no 'duplicate checksum' AC on the `replace` ticket: https://bugs.earthdata.nasa.gov/browse/CUMULUS-781

Reply:
You're right - I transposed `replace` and `version` in my head.

Reply:
For the version case, you're right that there is no checksumming performed in the unit test. It is something we cover extensively in integration tests. I'd like to have a discussion about whether to move some of that to unit tests (out of scope for this ticket imo), but I don't know that we need to test it again in these unit tests.

Reply:
Speaking of which, I don't see any test of that sort for `version`. Do you think that would be something worthwhile to add as a unit test?